--- "mxe/src/ffmpeg.mk"
+++ "mxe-spek/src/ffmpeg.mk"
@@ -9,9 +9,7 @@
 $(PKG)_FILE     := $(PKG)-$($(PKG)_VERSION).tar.bz2
 $(PKG)_URL      := https://ffmpeg.org/releases/$($(PKG)_FILE)
 $(PKG)_URL_2    := https://launchpad.net/ffmpeg/$(call SHORT_PKG_VERSION,$(PKG))/$($(PKG)_VERSION)/+download/$($(PKG)_FILE)
-$(PKG)_DEPS     := gcc bzip2 gnutls lame libass libbluray libbs2b libcaca \
-                   libvpx opencore-amr opus sdl speex theora vidstab \
-                   vo-amrwbenc vorbis x264 xvidcore yasm zlib
+$(PKG)_DEPS     := gcc pthreads bzip2 opus yasm zlib
 
 # DO NOT ADD fdk-aac OR openssl SUPPORT.
 # Although they are free softwares, their licenses are not compatible with
@@ -39,31 +37,24 @@
         --yasmexe='$(TARGET)-yasm' \
         --disable-debug \
         --enable-memalign-hack \
-        --disable-pthreads \
-        --enable-w32threads \
+        --enable-pthreads \
+        --disable-w32threads \
         --disable-doc \
+        --disable-programs \
         --enable-avresample \
         --enable-gpl \
         --enable-version3 \
         --extra-libs='-mconsole' \
-        --enable-avisynth \
-        --enable-gnutls \
-        --enable-libass \
-        --enable-libbluray \
-        --enable-libbs2b \
-        --enable-libcaca \
-        --enable-libmp3lame \
-        --enable-libopencore-amrnb \
-        --enable-libopencore-amrwb \
         --enable-libopus \
-        --enable-libspeex \
-        --enable-libtheora \
-        --enable-libvidstab \
-        --enable-libvo-amrwbenc \
-        --enable-libvorbis \
-        --enable-libvpx \
-        --enable-libx264 \
-        --enable-libxvid
+        --disable-avdevice \
+        --disable-swresample \
+        --disable-swscale \
+        --disable-postproc \
+        --disable-avfilter \
+        --disable-encoders \
+        --disable-muxers \
+        --disable-devices \
+        --disable-filters
     $(MAKE) -C '$(1)' -j '$(JOBS)'
     $(MAKE) -C '$(1)' -j 1 install
 endef

--- "mxe/src/wxwidgets.mk"
+++ "mxe-spek/src/wxwidgets.mk"
@@ -9,7 +9,7 @@
 $(PKG)_SUBDIR   := wxWidgets-$($(PKG)_VERSION)
 $(PKG)_FILE     := wxWidgets-$($(PKG)_VERSION).tar.bz2
 $(PKG)_URL      := https://$(SOURCEFORGE_MIRROR)/project/wxwindows/$($(PKG)_VERSION)/$($(PKG)_FILE)
-$(PKG)_DEPS     := gcc expat jpeg libiconv libpng sdl tiff zlib
+$(PKG)_DEPS     := gcc expat jpeg libiconv libpng tiff zlib
 
 define $(PKG)_UPDATE
     $(WGET) -q -O- 'https://sourceforge.net/projects/wxwindows/files/' | \
@@ -35,7 +35,6 @@
         --with-regex=yes \
         --with-zlib=sys \
         --with-expat=sys \
-        --with-sdl \
         --without-gtk \
         --without-motif \
         --without-mac \
